packages/react-dom/src/events/DOMModernPluginEventSystem.js

```diff
@@ -305,3 +306,39 @@ export function attachElementListener(listener: ReactDOMListener): void {
 export function detachElementListener(listener: ReactDOMListener): void {
   // TODO
 }
+
+export function accumulateTwoPhaseListeners(event: ReactSyntheticEvent): void {
+  const phasedRegistrationNames = event.dispatchConfig.phasedRegistrationNames;
+  if (phasedRegistrationNames == null) {
+    return;
+  }
+  const dispatchListeners = (event._dispatchListeners = []);
+  const dispatchInstances = (event._dispatchInstances = []);
+  const {bubbled, captured} = phasedRegistrationNames;
+  let node = event._targetInst;
+
+  // Accumulate all instances and listeners via the target -> root path.
+  // In order to reduce allocations, we store the listeners and instances
+  // in pairs. i.e. [func, <div>, func2, <span>, ...]
```

Comment:
Is the comment outdated? Seems like you're using two arrays now instead of one with pairs.

Reply:
Yeah, I fixed :)